fiftyone/utils/selection.py

```diff
@@ -0,0 +1,183 @@
+"""
+Utilities for selecting content from datasets.
+
+| Copyright 2017-2020, Voxel51, Inc.
+| `voxel51.com <https://voxel51.com/>`_
+|
+"""
+from collections import defaultdict
+import warnings
+
+from bson import ObjectId
+
+import fiftyone.core.fields as fof
+import fiftyone.core.labels as fol
+from fiftyone.core.expressions import ViewField as F
+
+
+def select_samples(sample_collection, sample_ids):
+    """Selects the specified samples from the collection.
+
+    Args:
+        sample_collection: a
+            :class:`fiftyone.core.collections.SampleCollection`
+        sample_ids: an iterable of sample IDs to select
+
+    Returns:
+        a :class:`fiftyone.core.view.DatasetView` containing only the specified
+        samples
+    """
+    return sample_collection.select(sample_ids)
+
+
+def exclude_samples(sample_collection, sample_ids):
+    """Excludes the specified samples from the collection.
+
+    Args:
+        sample_collection: a
+            :class:`fiftyone.core.collections.SampleCollection`
+        sample_ids: an iterable of sample IDs to exclude
+
+    Returns:
+        a :class:`fiftyone.core.view.DatasetView` that excludes the specified
+        samples
+    """
+    return sample_collection.exclude(sample_ids)
+
+
+def select_objects(sample_collection, objects):
+    """Selects the specified objects from the sample collection.
+
+    The returned view will omit samples, sample fields, and individual objects
+    that do not appear in the provided ``objects`` argument, which should have
+    the following format::
+
+        [
+            {
+                "sample_id": "5f8d254a27ad06815ab89df4",
```

Comment:
Does this also need `frame_number`?

Reply:
I didn't add that yet because filtering frame-level labels is not yet supported. I believe @benjaminpkane plans to work on that